File: Python/Objetos/Terminal/Terminal.py
'''
Created on 6 feb. 2019

@author: d18cralr
'''

class Terminal:
    
    
    '''
    Constructor
    '''
    def __init__(self,numero):
        self.__numeroTelefono=numero
        self.__tiempoLlamada=0
    
    @property #Se puede usar como un "getter"
    def numeroTelefono(self):
        return self.__numeroTelefono
    
    @numeroTelefono.setter #setter de lado
    def numeroTelefono(self):
        self.__numeroTelefono
        
    @property #Se puede usar como un "getter"
    def tiempoLlamada(self):
        return self.__tiempoLlamada
    
    @tiempoLlamada.setter #setter de lado
    def tiempoLlamada(self,tiempo):
        self.__tiempoLlamada=tiempo
    
    '''
    Método llamar
    @param t numero de teléfono
    @param tiempo tiempo de la conversación
    '''    
    def llamar(self,t,tiempo):
        self.__tiempoLlamada += tiempo #Acumular el tiempo total que ha hablado con un terminal
        t.tiempoLlamada+=tiempo #Acumular el tiempo que se le pasa a un terminal como parametro
    
    '''
    Método toString 
    '''
    def __str__(self):
        return f"nº:{self.__numeroTelefono}-{self.__tiempoLlamada} s."
 
 
 
      
if __name__=="__main__":
    
    t1=Terminal("111 111 111")
    t2=Terminal("222 222 222")
    t3=Terminal("333 333 333")
    
    print(t1)
    print(t2,"\n")
    print("Tiempo tras llamada: ")
    t1.llamar(t2,20)
    t1.llamar(t3,50)
    print(t1)
    print(t2)
    print(t3)
   
    

File: Java/Tema2-POO/Terminal/Terminal.java
/**
 * 
 */
package Terminal;

/**
 * Implementa la clase Terminal. Un terminal tiene asociado un número. Los
 * terminales se pueden llamar unos a otros y el tiempo de conversación corre
 * para ambos. A continuación se proporciona el contenido del main y el
 * resultado que debe aparecer por pantalla.
 *
 * @author Rafael Miguel Cruz Álvarez
 * @version 1.0
 */
public class Terminal {
  
  
  private String numTelefono;
  private int tiempoLLamada;

  public Terminal(String numero) {
    setNumTelefono(numero);
    this.tiempoLLamada=0;
  }

  /**
   * Método getter del número de teléfono
   * @return the numTelefono
   */
  public String getNumTelefono() {
    return numTelefono;
  }

  /**
   * Método Setter del número de teléfono
   * @param numTelefono the numTelefono to set
   */
  private void setNumTelefono(String numTelefono) {
    this.numTelefono = numTelefono;
  }
  

  /**
   * Método getter del tiempo de llamada
   * @return the tiempoLLamada
   */
  public int getTiempoLLamada() {
    return tiempoLLamada;
  }

  /**
   * * Método setter del tiempo de llamada
   * @param tiempoLLamada the tiempoLLamada to set
   */
  private void setTiempoLLamada(int tiempoLLamada) {
    this.tiempoLLamada = tiempoLLamada;
  }
  
  /**
   * Método llamada entre terminales
   * @param t objeto terminal
   * @param tiempo tiempo empleado en la llamada
   */
  
  public void llamar(Terminal t, int tiempo) {
    this.tiempoLLamada += tiempo; //Acumular el tiempo total que ha hablado con un terminal
    t.tiempoLLamada+=tiempo; //Acumular el tiempo que se le pasa a un terminal como parametro
  }

  /* (non-Javadoc)
   * @see java.lang.Object#toString()
   */
  @Override
  public String toString() {
    return "nº "+getNumTelefono()+" - "+getTiempoLLamada()+"s.";
  }
  
  
  
  
  
}
